### PR TITLE
Nested search filter + remove iql

### PR DESCRIPTION
**Pull request by gnusi**

### CODE REVIEW

**Review by MBkkt — COMMENTED** [review]
If you make:
1) reset --soft to branch without your commits
2) save list of names all changed files
3) git stash
4) format all was changed files
5) git stash pop resolve conflicts to stashed
6) commit

It will be more easy to review

File: core/search/nested_filter.cpp [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+  doc_id_t SeekInternal(doc_id_t parent) {
+    assert(!doc_limits::eof(parent));
+    auto& doc = std::get<document>(attrs_);
+
+    // First valid child
+    child_->seek(parent + 1);
+
+    do {
+      const auto child = child_doc_->value;
+      assert(child > parent_doc_->value);
```
Comment (Dronplane) [inline core/search/nested_filter.cpp:149]:
```suggestion
      assert(child >= parent_doc_->value);
```
Looks like original  assert  could be violated for NoneMatcher in the following way:
We have the following index with just 1 parent and 1 child document (and child matches the desired filter)
1   2
p   c

1. we call ByNestedFilter::seek(1)  
2. parent_->value is 1
3. called SeekInternal(1)
4. child_ is seeked to 2 so child_->value is 2
5. doc.value is set to 1
6. parent_->next() is called so parent_->value = eof
7. NoneMatcher::Accept is called
8. condition on line 236 is false as eof is greater than 2
9. child_ is seeked to eof e.g. child_->value is eof
10. cycle is repeated and here we have child = eof and parent = eof
BTW on next run cycle will exit and eof will be returned so in general this works as needed.

File: core/search/nested_filter.cpp [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+/*static*/ filter::ptr ByNestedFilter::make() {
+  return memory::make_unique<ByNestedFilter>();
+}
+
+filter::prepared::ptr ByNestedFilter::prepare(
+    const index_reader& rdr, const Order& ord, score_t boost,
+    const attribute_provider* ctx) const {
+  auto& [parent, child, match, merge_type] = options();
+
+  if (match.Max < match.Min || !parent || !child) {
```
[the remark this answers is not in the document]
Comment (gnusi) [reply core/search/nested_filter.cpp:565]: Nothing prevents user from doing that.

File: tests/search/sort_tests.cpp [hunk 3]
```diff
@@ -944,7 +1016,11 @@ TEST(ScoreFunction_test, reset) {
   ASSERT_TRUE(func);
   ASSERT_NE(nullptr, func.Func());
   ASSERT_EQ(nullptr, func.Ctx());
-  func(nullptr);  // noop bu default
```
[the remark this answers is not in the document]
Comment (gnusi) [reply tests/search/sort_tests.cpp:947]: I've added an assertion checking that argument isn't `nullptr` as calling `std::memset` with `nullptr` is UB

File: tests/store/directory_cleaner_tests.cpp [hunk 4]
```diff
@@ -37,6 +37,13 @@ namespace {
 
 using namespace irs;
 
+auto MakeByTerm(std::string_view name, std::string_view value) {
```
[the remark this answers is not in the document]
Comment (gnusi) [reply tests/store/directory_cleaner_tests.cpp:40]: Will do in the upcoming PR.

File: core/search/conjunction.hpp [hunk 5]
```diff
@@ -114,7 +114,7 @@ class conjunction : public doc_iterator, private Merger, private score_ctx {
         const_cast<document*>(front_doc_);
     std::get<attribute_ptr<cost>>(attrs_) = irs::get_mutable<cost>(front_);
 
-    if constexpr (HasScore<Merger>()) {
+    if constexpr (HasScore_v<Merger>) {
```
[the remark this answers is not in the document]
Comment (gnusi) [reply core/search/conjunction.hpp:117]: Will do in the upcoming PR.

File: core/index/iterators.cpp [hunk 6]
```diff
@@ -149,22 +135,19 @@ struct empty_column_reader final : irs::column_reader {
   // Returns the corresponding column iterator.
   // If the column implementation supports document payloads then it
   // can be accessed via the 'payload' attribute.
-  virtual irs::doc_iterator::ptr iterator(bool /*consolidtaion*/) const override {
+  virtual irs::doc_iterator::ptr iterator(irs::ColumnHint /*consolidtaion*/) const override {
```
[the remark this answers is not in the document]
Comment (gnusi) [reply core/index/iterators.cpp:138]: Will do in the upcoming PR.

[the remark this answers is not in the document]
Comment (gnusi) [reply core/index/iterators.cpp:138]: Will do in the upcoming PR.

**Pull request merged**